docs/source/traitsui_user_manual/scripts/regenerate_example_screenshots.py

```diff
@@ -0,0 +1,151 @@
+# (C) Copyright 2004-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+import contextlib
+import os
+from unittest import mock
+from functools import partialmethod
+
+import pkg_resources
+
+from traits.api import HasTraits
+from traitsui.testing.api import UITester
+
+# Demo files are part of the package data.
+DEMO = pkg_resources.resource_filename("traitsui", "examples/demo")
+
+
+def _is_python_file(path):
+    """ Return true if the given path is (public) non-test Python file."""
+    _, basename = os.path.split(path)
+    _, ext = os.path.splitext(basename)
+    return (
+        ext == ".py"
+        and not basename.startswith("_")
+        and not basename.startswith("test_")
+    )
+
+
+def get_python_files(directory):
+    """ Report Python files to be tested or to be skipped.
+
+    Returns
+    -------
+    accepted_files : list of str
+        Python file paths to be tested.
+    """
+    accepted_files = []
+    for root, _, files in os.walk(directory):
+        for filename in files:
+            path = os.path.abspath(os.path.join(root, filename))
+            if not _is_python_file(path):
+                continue
+            accepted_files.append(path)
+
+    # Avoid arbitrary ordering from the OS
+    return sorted(accepted_files)
+
+
+def get_editor_example_filees(python_files):
+    Editor_demo_files = []
+    for filename in python_files:
+        if filename.endswith('Editor_demo.py'):
+            Editor_demo_files.append(filename)
+    return Editor_demo_files
+
+
+def replaced_configure_traits(
+    instance,
+    filename=None,
+    view=None,
+    kind=None,
+    edit=True,
+    context=None,
+    handler=None,
+    id="",
+    scrollable=None,
+    screenshot_name=None,
+    **args,
+):
+    """ Mocked configure_traits to launch then close the GUI.
+    """
+    ui_kwargs = dict(
+        view=view,
+        parent=None,
+        kind="live",  # other options may block the test
+        context=context,
+        handler=handler,
+        id=id,
+        scrollable=scrollable,
+        **args,
+    )
+    with UITester().create_ui(instance, ui_kwargs) as ui:
+        ui_pixmap = ui.control.grab()
+        ui_pixmap.save(
+            os.path.join(
+                "docs",
+                "source",
+                "traitsui_user_manual",
+                "images",
+                screenshot_name
+            )
```

Comment:
looks like this is the reason why this script needs to be run from the root of the directory.

like we discussed over chat, we can simply document the fact that this script needs to be run from the root.